### ISSUE TITLE
Allow linking to Bibliography Elements with `#link(<citation>)`

### ISSUE THREAD

**Issue by timon-schelling** [typst/typst#3367]
### Description

Bibliography Elements cannot be referenced via `#link(<citation>)`, only labels.
My suggestion is that `link` links to Bibliography Elements similar to `ref` and `cite` but with custom text.

### Use Case

Link to a Bibliography Elements with custom text.
```
#link(<citation>)[Some content]
```
Seems like a common usecase.

Currently, one has to use a csl file that only displays the supplement to active the same result.
```
#let link_to_bibliography_element(key, text) = {
  cite(key, supplement: text, style: "only-supplement.csl")
}
```
```
<citation>
  <layout>
    <text variable="locator"/>
  </layout>
</citation>
```

**Comment by Enivex** [comment]
It seems to me like what you want is really https://github.com/typst/typst/issues/1139

**Comment by timon-schelling** [comment]
It would be possible to archive what I want easier, yes. But regardless as a new user I would still expect that it is possible to link to every label including bib elements.

**Comment by leonmartin** [comment]
I agree with @Enivex! In my case, I want to create custom symbols that link to bibliography entries. I also think there are lots of other use cases for linking bibliography entries.

@Enivex , what is the exact csl file you are using for your workaround?

**Comment by timon-schelling** [comment]
@leonmartin I use it in this repo https://github.com/timon-schelling/uni-phi111-essay-2024-02-15/tree/main/src hit me up if you need help getting it to work :)

**Comment by leonmartin** [comment]
@timon-schelling Thank you very much, works great!
